### PR TITLE
Developer documentation 

### PR DESCRIPTION
**Pull request by MichielTukker**
First attempt at developer documentation.  

I've reviewed the scope and process of the docs,  which are good enough for now.   The documentation describes all requested parts (architecture, quick how-to's,  testing,  documentation,  api-doc). 

This PR also includes a new agent:  RstSyntaxAgent   it's goal is to validate the RST syntax as I had some issues with the sphinx validation agent.  The RstSyntaxAgent calls some specific Sphinx commands for Rst validation,  so it should be reliable.

### CODE REVIEW

File: doc/developer/developer_main.rst [hunk 1]
```diff
@@ -24,8 +32,10 @@ How to use this section
    :titlesonly:
 
    developer_navigation
-   ../reference/architecture_reference
-   ../reference/asset_reference
-   ../reference/solver_reference
-   ../reference/controller_reference
+   simulation_contributor_flow
```
[the remark this answers is not in the document]
Comment (MichielTukker) [reply doc/developer/developer_main.rst]: These files will be overwritten by other contributors

File: doc/developer/add_new_output_format.rst [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+- :doc:`add_new_asset` — complete workflow for adding a new asset type, including
+  entity-layer, solver-layer, ESDL mapper, and mapper registration steps that precede
+  output implementation for a brand-new asset.
+- :doc:`simulation_contributor_flow` — end-to-end code layout and execution flow from
+  ``app.py`` through ``SimulationManager``, ``NetworkSimulation``, and ``HeatNetwork``.
+- :doc:`testing` — test layout, how to run the full test suite, and the contributor
+  checklist.
+- :doc:`../architecture/network_simulation` — conceptual overview of how the network
+  simulation coordinates assets, the solver, and the controller.
```
[the remark this answers is not in the document]
Comment (MichielTukker) [reply doc/developer/add_new_output_format.rst]: Will be written in another PR

File: .github/instructions/documentation-architecture.instructions.md [hunk 3]
```diff
@@ -54,6 +54,15 @@ Use instead:
...
 - simple fractions/division where clearer.
 
+Code examples and snippets
+--------------------------
+When including code examples or snippets:
+- Use the repository's actual code and file structure where possible.
+- Use the repository's actual class and function names where possible.
+- Use the repository's actual variable names where possible.
+
+When using a code block to show a commandline invocation, ensure that the command is valid for both Windows and Linux shells. Use a single code block with a single command, and do not include separate Windows/Linux blocks unless the command is actually different between platforms. Use the simplest form of the command possible, and do not include extra flags or options that are not required for the example to work. If the command is different between platforms, explain the difference in text instead of using separate code blocks.
```
[the remark this answers is not in the document]
Comment (MichielTukker) [reply .github/instructions/documentation-architecture.instructions.md]: modified text

File: .vscode/settings.json [hunk 4]
```diff
@@ -21,22 +21,7 @@
...
-    },
     ".\\doc\\run_spinx.bat": true,
-    "/^& \\.\\\\venv\\\\Scripts\\\\python\\.exe -m sphinx -b html doc doc\\\\_build\\\\html$/": {
-      "approve": true,
-      "matchCommandLine": true
-    }
   },
-}
+  "python-envs.defaultEnvManager": "ms-python.python:system",
+}
```
[the remark this answers is not in the document]
Comment (MichielTukker) [reply .vscode/settings.json:27]: Fixed

File: doc/developer/AddEntityLayerAsset.rst [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+Inherit from ``AssetAbstract`` and assign the existing solver-layer asset in ``__init__``:
+
+.. code-block:: python
+
+    from omotes_simulator_core.entities.assets.asset_abstract import AssetAbstract
+    from omotes_simulator_core.solver.network.assets.existing_solver_asset import ExistingSolverAsset
+
+    class MyAssetVariant(AssetAbstract):
+        asset_type = "my_asset_variant"
```
Comment (vanmeerkerk) [inline doc/developer/AddEntityLayerAsset.rst]: Waarom dit hier? Als ik naar andere assets kijk wordt dit niet gedaan. Een standaard asset heeft altijd twee punten dus hoeft niet gedefinieerd te worden; verwijs ze naar AssetAbstract om duidelijk te krijgen wat er moet worden ingevuld.

File: doc/developer/AddOutputFormat.rst [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+
+To validate the documentation build after updating this page, use the repository
+convenience script. This is an intentional platform-specific exception because the
+wrapper script differs by shell (``.bat`` on Windows and ``.sh`` on Linux):
+
+.. code-block:: console
+
+  doc/run_spinx.bat
+
+On Linux, run ``doc/run_sphinx.sh`` instead.
```
Comment (vanmeerkerk) [inline doc/developer/AddOutputFormat.rst]: Is this necessary in every rst file or should it be placed in an overhead file?

Comment (vanmeerkerk) [inline doc/developer/AddOutputFormat.rst]: Is this necessary in every rst file or should it be placed in an overhead file?

File: doc/developer/ControlExtension.rst [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+Edit ``src/omotes_simulator_core/entities/assets/controller/controller_network.py``.
+
+In the ``ControllerNetwork.__init__`` constructor or in ``EsdlControllerMapper``, add the new
+asset controller to the appropriate list:
+
+- ``self.producers`` — for production assets
+- ``self.consumers`` — for demand assets
+- ``self.storages`` — for storage assets
+- ``self.heat_transfer_assets_prim`` or ``self.heat_transfer_assets_sec`` — for heat exchangers
+  or heat pumps
```
Comment (vanmeerkerk) [inline doc/developer/ControlExtension.rst]: Misschien duidelijker maken dat dit assets zijn tussen twee netwerken?

File: doc/developer/documentation_with_ai_agents.rst [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+This guide explains how contributors use the SIMULATOR-CORE AI agent workflow to write,
+review, and validate documentation. It is written for anyone who needs to add or modify
+documentation in this project and wants to understand which agents exist, what each one
+does, how to invoke them, and how to validate the result.
+
+Use this guide instead of the conceptual documentation or the API reference when your goal
+is to write documentation itself — new pages, corrections to existing pages, or navigation
+updates.
+
+Relevant Agent Files
```
Comment (vanmeerkerk) [inline doc/developer/documentation_with_ai_agents.rst]: How do we ensure this list of agents stays up to date?

File: doc/developer/architecture.rst [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    constraints — they do not know about time series or ESDL properties.
+
+Each entity-layer asset creates its corresponding solver-layer asset in its constructor and
+registers it with the ``Network`` graph object. The entity layer reads solver results through
+``self.solver_asset`` after each ``Solver.solve()`` call.
+
+The dependency direction is one-way: the entity layer depends on the solver layer. Solver-layer
+classes must never import from ``entities/``.
+
+Control Architecture
```
Comment (samvanderzwan) [inline doc/developer/architecture.rst]: I am missing short description that user can also add its own control be adding a new Networkcontroller class which inheritant from the abstract class. 
Also the first line it does not sits between. It defines the setpoints for the heatnetwork.

File: doc/developer/architecture.rst [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+--------------------
+
+The control layer sits between ``NetworkSimulation`` and ``HeatNetwork``:
+
+- ``NetworkControllerAbstract`` defines the contract: one abstract method,
```
Comment (samvanderzwan) [inline doc/developer/architecture.rst]: What is meant with contract?

File: doc/developer/simulation_flow.rst [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+   * - ``src/omotes_simulator_core/adapter/transforms/controller_mapper.py``
+     - ESDL-to-controller mapper
+     - ``EsdlControllerMapper.to_entity`` converts an ``EsdlObject`` into a
+       ``NetworkController`` containing the correct ``ControllerNetwork`` objects.
+   * - ``src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py``
+     - Asset dispatch table
+     - Maps ESDL asset types to per-asset mapper classes via ``conversion_dict_mappers``.
+   * - ``unit_test/infrastructure/test_simulationmanager.py``
+     - Orchestration unit tests
+     - Verifies ``SimulationManager.execute`` returns output and triggers the progress
```
Comment (samvanderzwan) [inline doc/developer/simulation_flow.rst]: Why add the unit test here? and only a couple

File: doc/developer/testing_strategy.rst [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+- Use ``Mock(spec=ControllerNetwork)`` for the network argument.
+- Call ``update_setpoints`` and assert that the returned dict contains the expected keys and
+  value types.
+- Test edge cases such as zero demand, empty subnetworks, or profile boundaries.
+
+Adapter tests
+~~~~~~~~~~~~~
+
+Mapper tests live in ``unit_test/adapters/``. The main file is
+``unit_test/adapters/test_esdl_graph_mapper.py``. Mapper tests verify that a known ESDL
```
Comment (samvanderzwan) [inline doc/developer/testing_strategy.rst]: File location is incorrect as that this is the main file

File: doc/developer/AddEntityLayerAsset.rst [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+        number_of_con_points: int = 2  # adjust to match port count
+
+        def __init__(self, asset_name: str, asset_id: str, connected_ports: list[str]) -> None:
+            super().__init__(asset_name, asset_id, connected_ports)
+            self.solver_asset = ExistingSolverAsset(
+                name=asset_name,
+                _id=asset_id,
+            )
+            self._register_solver_asset(self.solver_asset)
```
Comment (samvanderzwan) [inline doc/developer/AddEntityLayerAsset.rst]: This method does not exist at all

File: doc/developer/AddSolverAndEntityLayerAssets.rst [hunk 14]
```diff
@@ -0,0 +1,230 @@
+Adding New Solver and Entity Layer Assets
```
Comment (samvanderzwan) [inline doc/developer/AddSolverAndEntityLayerAssets.rst]: This is partly a repeat of Add ENtityLayerAsset. So better to reference to that

File: doc/developer/ControlExtension.rst [hunk 15]
```diff
@@ -0,0 +1,290 @@
...
+    (profiles, demand, stored energy, COP) and the logic to compute per-asset setpoints.
+
+**Network-level controller** (``entities/network_controller.py``)
+    The ``NetworkController`` class. It owns a list of ``ControllerNetwork`` objects
+    (one per hydraulically isolated sub-network). Each ``ControllerNetwork`` contains the
+    asset-level controllers for its sub-network. ``NetworkController`` orchestrates the
+    dispatch: update network state, calculate per-network totals, distribute demand across
+    producers, and return the setpoints dict.
+
+**Abstract contract** (``entities/network_controller_abstract.py``)
```
Comment (samvanderzwan) [inline doc/developer/ControlExtension.rst]: Would not call it contract

**Review by samvanderzwan — CHANGES_REQUESTED** [review]
